python/CHANGELOG.rst

```diff
@@ -2,6 +2,12 @@
 [0.5.4] - 2022-XX-XX
 --------------------
 
+**Features**
+ - The ``Tree`` object now has the method ``siblings`` to get
+   the siblings of a node. It returns an empty tuple if the node
+   has no siblings, is not a node in the tree, is a virtual root,
+   or is isolated. (:user:`szhan`, :pr:`2618`, :issue:`2616`)
```

Comment:
I don't think it will return an empty tuple for isolated nodes, right? They are roots, so `siblings()` will return the other roots

Reply:
An isolated sample node is a root, but an isolated non-sample node is not a root?

```
tables = tskit.Tree.generate_balanced(2, arity=2).tree_sequence.dump_tables()
tables.nodes.add_row(flags=0)   # Add node 3
t = tables.tree_sequence().first()
display.SVG(t.draw_svg())
```

`t.roots` returns `[2]`.

But when I set `flags=1`, `t.roots` returns `[3, 2]`.

So, it should really depend on whether the isolated node is a sample node or not?

Reply:
Wait... when `flags=0`, `t.children(t.virtual_root)` returns `(2,)`, But when `flags=1`, that returns `(3, 2)` instead.

Reply:
Yes.

"However, it is also possible for a [sample node](https://tskit.dev/tskit/docs/stable/glossary.html#sec-data-model-definitions-sample) to be isolated. Unlike other nodes, isolated sample nodes are still considered as being present on the tree (meaning they will still returned by the [Tree.nodes()](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.Tree.nodes) and [Tree.samples()](https://tskit.dev/tskit/docs/stable/python-api.html#tskit.Tree.samples) methods): they are therefore plotted, but unconnected to any other nodes."

(https://tskit.dev/tskit/docs/stable/data-model.html#isolated-nodes)

Reply:
So when an isolated node is a *sample* node, it is considered still present in the tree, as another root.